book/src/volume-expansion.md

```diff
@@ -0,0 +1,46 @@
+# Volume Expansion
+
+**Status:** Alpha
+
+Many storage providers allow a volume to be expanded after creation. A storage provider may choose to implement volume expansion either via a
+control-plane CSI RPC call or via node CSI RPC call or both as a two step process.
+
+## Implementing Volume expansion functionality
+
+To implement volume expansion the CSI driver MUST:
+
+1. Implement `VolumeExpansion` plugin capability.
+2. Implement `EXPAND_VOLUME` controller capability or implement `EXPAND_VOLUME` node capability or both.
+
+`ControllerExpandVolume` RPC call can be made when volume is `ONLINE` or `OFFLINE` depending of `VolumeExpansion` plugin
+capability.
+
+`NodeExpandVolume` RPC call on the other hand - *always* requires volume to be published or staged on a node (and hence `ONLINE`).
+For block storage file systems - `NodeExpandVolume` is typically used for expanding the file system on the node, but it can be also
+used to perform other volume expansion related housekeeping operations on the node.
+
+For details,  see the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md).
+
+## Deploying volume expansion functionality
+
+The Kubernetes CSI development team maintains [external-resizer](external-resizer.md) Kubernetes CSI [Sidecar Containers](sidecar-containers.md).
+This sidecar container implements the logic for watching the Kubernetes API for Persistent Volume claim edits and issuing the appropriate CSI expansion calls against a CSI endpoint. For more details, see [external-resizer](external-resizer.md).
```

Comment:
only the ControllerExpandVolume call right?

Reply:
yeah. I will call that out explicitly.